frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -267,9 +267,9 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 							#frame_support::sp_tracing::enter_span!(
 								#frame_support::sp_tracing::trace_span!(stringify!(#fn_name))
 							);
-							// We execute all dispatchable in at least one storage layer, allowing them
+							// We execute all dispatchable in a new storage layer, allowing them
 							// to return an error at any point, and undoing any storage changes.
-							#frame_support::storage::in_storage_layer(|| {
+							#frame_support::storage::with_storage_layer(|| {
```

Comment:
This change does not apply to testing, or?  
So now when an extrinsic is called from testing it behaves differently as when its dispatched.  
The test case from the linked finding sill fails.

Reply:
this was already an issue with testing. See the developer notes here: https://github.com/paritytech/substrate/pull/11431

Reply:
Okay so its not worse than before, since the tests already assumed `in_storage_layer`. :+1: